### PR TITLE
Footer

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #31 
closes https://github.com/compilerla/compiler.la/issues/38

## What this PR does
- **Footer**: Change fonts and spacing as necessary for new design
- **Header**: Standardizing the header margins matters for the Footer, b/c the Footer and Header must be left-aligned together. This PR uploads a new compiler logo, with flush edges, so we don't have to use negative margin CSS to remove the random space between the logo and the edge of the image.
- **CSS**: Starts to incorporate CSS Variables and Bootstrap variable overrides for Footer and Header changes.
- **Container:** All pages have `container-fluid` so we don't have this wildly wide page issue https://github.com/compilerla/compiler.la/issues/38
- **Container**: This PR removes a lot of the alignment/flex styling that was put on `body`. Instead, it uses a `--bs-gutter-x` override to declare that on Desktop, the gutter of the main `container-fluid` should be 24px on each side, and 16px on each side for Mobile, to go along with what is on Figma. As you can see from the Figma screenshots below, the x gutter should be 24px on Desktop and 16px on mobile. This change is declared here: https://github.com/compilerla/compiler.la/pull/106/files#diff-dd51c5cd06799351fdf26f10840213ef27390c39ed033c1a7d82ac493f8eb696R26-R34

image
image

## Screenshots

Figma vs. this PR:
image
image
image
image

Ensure no regressions:
image
image

### CODE REVIEW

**Comment by machikoyasuda** [comment]
@angela-tran @thekaveman This is ready for review. Note that this is for the Footer _only_. Home Page fix PR is up next, so we won't be in limbo purgatory for too long with the footer looking different from the home page.

https://www.figma.com/file/2gq3WJFdRFDefyHcgFMUGt/Current-compiler.la?node-id=1355%3A3984&mode=dev *Just* the lower half of this page, the top of this page is to look just like compiler.la production

**Comment by machikoyasuda** [comment]
Thanks to @segacy1 for sending a better tightly-cropped SVG logo!

**Comment by machikoyasuda** [comment]
Ready for review @angela-tran

File: styles/base.css [hunk 1]
```diff
@@ -204,22 +176,15 @@ h6 {
...
-        --header-margin-left: -10px;
         --logo-lg-width: 20rem;
     }
 }
 
-header {
-    margin-left: var(--header-margin-left);
-    /* account for padding within the Compiler logo*/
-}
-
```
Comment (machikoyasuda) [inline styles/base.css:222]: This is no longer necessary, b/c the logo file is updated to be tightly cropped.

File: styles/base.css [hunk 2]
```diff
@@ -8,49 +8,39 @@
...
-    flex-direction: column;
-
-    padding: var(--body-padding);
-    background-color: var(--brand-primary-black);
-
-    color: var(--white);
-    font-family: 'Roboto', Arial, Helvetica, sans-serif;
-    font-weight: normal;
-}
-
```
Comment (machikoyasuda) [inline styles/base.css:44]: Using the `--bs-gutter-x` to set body padding. So this is no longer necessary. This is how the override looks like in the CSS:

image

The default `--bs-gutter-x`(left and right) is 1.5rem, but I'm setting it to 3rem/2rem (48px total/32px total responsively across Desktop/Mobile). 

Here's the 48px gutter on Desktop in effect. There is 24px on each side:
image

File: styles/base.css [hunk 3]
```diff
@@ -63,40 +53,22 @@ a:focus {
...
-.jobs li {
-    margin: 1rem 0;
-}
-
-@media (min-width: 992px) {
-    .jobs li {
-        margin: 0.5rem 0 0;
-    }
-}
-
```
Comment (machikoyasuda) [inline styles/base.css:85]: This is all managed with Bootstrap classes in `cert.html` instead.

File: styles/base.css [hunk 4]
```diff
@@ -228,92 +193,32 @@ header {
...
-footer .certs>div {
-    margin: 30px 0 0;
-    padding-left: var(--certs-div-padding-left);
-}
-
-footer .heading {
-    font-size: 1.5rem;
```
Comment (machikoyasuda) [inline styles/base.css:271]: All managed with Bootstrap utility classes in the template now.

File: _layouts/default.html [hunk 5]
```diff
@@ -49,31 +49,32 @@
...
-    <footer>
-        <div class="divider-line"></div>
-        <div class="certs d-flex flex-column flex-lg-row row-cols-lg-4">
-            <div>
-                <a href="mailto:hello@compiler.la" class="monospace contact-link">hello@compiler.la</a>
-                <p class="address">
-                    525 S Hewitt St<br />Los Angeles, CA 90013
-                </p>
-            </div>
+    <footer class="border-top border-light">
```
Comment (machikoyasuda) [inline _layouts/default.html:56]: The way this `border-light` Bootstrap class is getting its color is from the CSS root variable declaration here:

```css
    --bs-light-rgb: 84, 84, 84; /* for the HR above Footer */
    --bs-light: #545454; /* brand/scales/gray/4 */
```

More on `border-X` classes (like `border-primary`, `border-secondary`): https://getbootstrap.com/docs/5.2/utilities/borders/#color 
This also gives us the class: `text-light`: https://getbootstrap.com/docs/5.2/utilities/colors/#colors

File: _layouts/default.html [hunk 6]
```diff
@@ -49,31 +49,32 @@
...
+        <div class="container-fluid">
+            <div class="row d-flex flex-column flex-lg-row row-cols-lg-4">
+                <div class="pt-lg-5 mt-lg-1 pt-4 mt-3 pb-5">
+                    <a href="mailto:hello@compiler.la" class="monospace text-primary contact-link">hello@compiler.la</a>
```
Comment (machikoyasuda) [inline _layouts/default.html:60]: This `text-primary` class gets its color from this CSS Variable declaration:

```css
    --bs-primary: #88B440; /* Use with .text-primary */
    --bs-primary-rgb: 136, 180, 64;
```

Can also be used with `border-primary`.

File: styles/base.css [hunk 7]
```diff
@@ -228,92 +194,30 @@ header {
...
     transition: 250ms;
-    margin: 0 12px;
-    width: calc(28rem / 16);
+    width: calc(24rem / 16);
 }
 
 @media (min-width: 992px) {
     .social-media-icon {
-        margin: 0 24px;
-        width: 3rem;
```
Comment (machikoyasuda) [inline styles/base.css:296]: For the social media icon margins, I used the Bootstrap `gap` class instead:

`<div class="d-flex py-4 gap-md-4 gap-3">`

`gap-3` sets the children apart by 16px, and `gap-md-4` sets the children apart by 24px at medium/large screen sizes. Adding `d-flex` allows you to use gap classes.

More on gap: https://getbootstrap.com/docs/5.3/utilities/spacing/#gap

File: _layouts/default.html [hunk 8]
```diff
@@ -49,31 +49,32 @@
...
-        <div class="d-flex social-media-container">
-            <div class="d-flex justify-content-between align-items-end py-4">
-                <a href="https://github.com/compilerla"><img class="social-media-icon"
-                        src="/assets/social_media/github_mark.png" alt="GitHub"></a>
-                <a href="https://www.linkedin.com/company/compiler-la"><img class="social-media-icon"
-                        src="/assets/social_media/linkedin_in.png" alt="LinkedIn"></a>
             </div>
-            <img class="brandmark" src="/assets/compiler_brandmark.svg" alt="Compiler brandmark">
+            <div class="d-flex social-media-container">
+                <div class="d-flex py-4 gap-md-4 gap-3">
```
Comment (machikoyasuda) [inline _layouts/default.html:70]: See note about `gap` below.

File: styles/base.css [hunk 9]
```diff
@@ -1,56 +1,47 @@
...
 
 :root {
-    --purple: #663190;
     --green: #88B440;
-    --teal: #01B4CB;
-    --navy: #223061;
-    --brand-primary-black: #1C1C1C;
-    --black: #000000;
     --white: #FFFFFF;
+    --brand-primary-cyan: #51B1C7;
```
Comment (machikoyasuda) [inline styles/base.css:6]: `teal` is now called `cyan`, and the color changed from `#01B4CB` to `#51B1C7`

The color is called `cyan` on Figma, so I'm changing it to `cyan` here as well.

**Comment by machikoyasuda** [comment]
@angela-tran Okay, _now_, it's finally ready for review. I added informational comments on various files and lines.

File: styles/base.css (hunk 2 above)
Comment (machikoyasuda) [reply styles/base.css:44]: This `bs-gutter-x` variable is used not just here, but in any class concerning container and column widths: 
image

So it's pretty cool that with CSS Variables, we can now make cascading variable changes across the app without Sass.

File: styles/base.css [hunk 10]
```diff
@@ -228,92 +194,30 @@ header {
...
-    border-top: 1px solid #545454;
-    margin: 10px -16px 10px;
-}
-
-@media (min-width: 992px) {
-    .divider-line {
-        margin: 20px -30px 20px;
-    }
-
-}
```
Comment (machikoyasuda) [inline styles/base.css:249]: This is now `border-light border-top` using `bs-light-rgb` color. See comment above.

**Comment by machikoyasuda** [comment]
@thekaveman Hrm.. but `container` would make the header logo go too far in:

This is what `container` looks like for me:

image

and we want the logo to be aligned vertically with the Footer like this:
image

**Comment by thekaveman** [comment]
You're looking at a different branch I think? It looks like the homepage updates are there. 

The issue is with `container-fluid` everywhere, it expands to the whole width:

> `.container-fluid`, which is `width: 100%` at all breakpoints

https://getbootstrap.com/docs/5.2/layout/containers/

All 3 places it was added in this PR should be `container` I think?

**Comment by machikoyasuda** [comment]
@thekaveman But `container` would make the whole app show up with way too much margin left and right on 13 inch laptop size screens:

image

**Comment by thekaveman** [comment]
@machikoyasuda isn't that controlled by the Bootstrap gutter?

This doesn't look problematic to me? 1024px with `container` shows the standard 32px of margin on each side of all 3 elements

image

File: styles/base.css [hunk 11]
```diff
@@ -33,6 +32,12 @@ body {
     }
 }
 
+@media (min-width: 1920px) {
+    .container-fluid {
+        max-width: 1320px;
+    }
+}
```
Comment (machikoyasuda) [inline styles/base.css:39]: I got the 1920px width breakpoint suggestion from CSS-Tricks: https://css-tricks.com/optimizing-large-scale-displays/#aa-build-with-relative-units They have breakpoints at 1920px, 2560px. Bootstrap breakpoints end with the largest at 1400px.

I got the 1320px width from Bootstrap: https://getbootstrap.com/docs/5.2/layout/containers/ It's the max-width for containers of screens up to 1400px.

Comment (machikoyasuda) [reply styles/base.css:39]: @thekaveman Can you try this on your machine? Does 1320px feel like a good width, or should it be wider?

**Comment by thekaveman** [comment]
Yeah this looks a lot better, thank you! 🚀 

image

**Comment by thekaveman** [comment]
Side note: I think we could use a little bit more top spacing? Like another 0.5rem - 1rem or something

**Comment by machikoyasuda** [comment]
@thekaveman Yea but that will all be changed in #107 so we could just wait for that. Or I could add a class there for more spacing that would be on prod for a day or something.

**Comment by thekaveman** [comment]
> Yea but that will all be changed in https://github.com/compilerla/compiler.la/pull/107 so we could just wait for that.

All good

**Review by thekaveman — APPROVED** [review]

**Pull request merged**